pkg/webhook/admission/localmodelcache/local_model_cache_validation.go

```diff
@@ -0,0 +1,109 @@
+	"github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	"github.com/kserve/kserve/pkg/constants"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var (
+	// logger for the validation webhook.
+	localModelCacheValidatorLogger = logf.Log.WithName("localmodelcache-v1alpha1-validation-webhook")
+)
+
+// +kubebuilder:object:generate=false
+// +k8s:deepcopy-gen=false
+// +k8s:openapi-gen=false
+// LocalModelCacheValidator is responsible for validating the LocalModelCache resource
+// when it is created, updated, or deleted.
+//
+// NOTE: The +kubebuilder:object:generate=false and +k8s:deepcopy-gen=false marker prevents controller-gen from generating DeepCopy methods,
+// as this struct is used only for temporary operations and does not need to be deeply copied.
+type LocalModelCacheValidator struct {
+	client.Client
+}
+
+// +kubebuilder:webhook:verbs=delete,path=/validate-localmodelcaches,mutating=false,failurePolicy=fail,groups=serving.kserve.io,resources=localmodelcaches,versions=v1alpha1,name=localmodelcache.kserve-webhook-server.validator
+var _ webhook.CustomValidator = &LocalModelCacheValidator{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (v *LocalModelCacheValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	localModelCache, err := convertToLocalModelCache(obj)
+	if err != nil {
+		localModelCacheValidatorLogger.Error(err, "Unable to convert object to LocalModelCache")
+		return nil, err
+	}
+	localModelCacheValidatorLogger.Info("validate create", "name", localModelCache.Name)
+	return nil, nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (v *LocalModelCacheValidator) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (admission.Warnings, error) {
+	localModelCache, err := convertToLocalModelCache(newObj)
+	if err != nil {
+		localModelCacheValidatorLogger.Error(err, "Unable to convert object to LocalModelCache")
+		return nil, err
+	}
+	localModelCacheValidatorLogger.Info("validate update", "name", localModelCache.Name)
+	return nil, nil
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (v *LocalModelCacheValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	localModelCache, err := convertToLocalModelCache(obj)
+	if err != nil {
+		localModelCacheValidatorLogger.Error(err, "Unable to convert object to LocalModelCache")
+		return nil, err
+	}
+	localModelCacheValidatorLogger.Info("validate delete", "name", localModelCache.Name)
+
+	// Check if current LocalModelCache is being used
+	for _, isvcMeta := range localModelCache.Status.InferenceServices {
```

Comment:
Is it sufficient to check the size of the InferenceServices array assuming this status field is the "used isvcs"?

Reply:
@yuzisun This is more accurate. Is there a performance concern? e.g. if the model cache controller was not able to update the status field for some reason. like a bug.

Reply:
are you saying for some reason the used inference services are not removed from status?

Reply:
for example, the local model controller keep crashing, then maybe the service was not added to or removed from the status field. But yeah, not a big difference.

Reply:
is the concern there might be isvc using this not in the array? imo inactive isvc not removed from the array is more likely